addons/hr_holidays/report/hr_leave_employee_report.py

```diff
@@ -0,0 +1,148 @@
+                    WHERE hl.employee_company_id {f"IN {tuple(self.env.companies.ids)}" if len(self.env.companies.ids) > 1 else f"= {self.env.companies.id}"}
+                )
+                SELECT
+                    id, leave_id, employee_id,
+                    holiday_status_id, state, description,
+                    GREATEST(date_from, day_start) AS day_aligned_date_from,
+                    LEAST(date_to, (day_start + INTERVAL '1 day' - INTERVAL '1 second')) AS day_aligned_date_to
+                FROM leave_data;
+        """)
+        fetched_leave_field_names = [desc[0] for desc in self._cr.description]
+        leave_records = self._cr.fetchall()
+        return fetched_leave_field_names, leave_records
+
+    def _compute_is_hatched(self, report_records):
+        for report_record in report_records:
+            report_record['is_striked'] = report_record['state'] == 'refuse'
+            report_record['is_hatched'] = report_record['state'] not in ['refuse', 'validate']
+
+    def _create_report_records_from_leave_records(self, leave_records, leave_field_names):
+        report_records = [
+            {field_name: leave_record[index] for index, field_name in enumerate(leave_field_names)}
+            for leave_record in leave_records
+        ]
+        return report_records
+
+    def _compute_working_schedule_aligned_dates(self, report_records):
+        for report_record in report_records:
+            start_date = report_record['day_aligned_date_from'].replace(tzinfo=utc)
+            end_date = report_record['day_aligned_date_to'].replace(tzinfo=utc)
+            work_intervals = self.env['hr.leave'].browse(report_record['leave_id'])\
+                .resource_calendar_id._work_intervals_batch(start_date, end_date, compute_leaves=False)[False]\
+                .items()
+            if work_intervals:
+                report_record['working_schedule_aligned_date_from'] = work_intervals[0][0].astimezone(utc).replace(tzinfo=None)
+                report_record['working_schedule_aligned_date_to'] = work_intervals[-1][1].astimezone(utc).replace(tzinfo=None)
+            else:
+                # Fallback if no work intervals found
+                report_record['working_schedule_aligned_date_from'] = start_date.replace(tzinfo=None)
+                report_record['working_schedule_aligned_date_to'] = end_date.replace(tzinfo=None)
+
+    def _compute_leave_duration(self, report_records):
+        for report_record in report_records:
+            # Remove the day_aligned_date_from/to as they are intermediate values for computing other values
+            virtual_leave = self.env['hr.leave'].new({
+                'date_from': report_record.pop('day_aligned_date_from'),
+                'date_to': report_record.pop('day_aligned_date_to'),
+                'employee_id': report_record['employee_id'],
+                'holiday_status_id': self.env['hr.leave'].browse(report_record['leave_id']).holiday_status_id.id,
+            })
```

Comment:
As we try to avoid queries inside loops, I think extracting browse like this would be better for performance
```suggestion
        leave_ids = [report_record['leave_id'] for report_record in report_records] 
        leaves = self.env['hr.leave'].browse(leave_ids)
        holiday_status_id_by_leave_id = {leave.id: leave.holiday_status_id.id for leave in leaves}
        for report_record in report_records:
            # Remove the day_aligned_date_from/to as they are intermediate values for computing other values
            virtual_leave = self.env['hr.leave'].new({
                'date_from': report_record.pop('day_aligned_date_from'),
                'date_to': report_record.pop('day_aligned_date_to'),
                'employee_id': report_record['employee_id'],
                'holiday_status_id': holiday_status_id_by_leave_id[report_record['leave_id']],
            })
```